### DISCUSSION TITLE
Can this package be used to transfer OpenSea matic tokens?

### DISCUSSION THREAD

**Discussion by imkane** [0xPolygon/matic.js#272] [Q&A] [answered]
I tried using (a customized version of) the code in https://github.com/maticnetwork/matic.js/blob/master/examples/plasma/erc721/transfer.js to attempt to transfer this token:

https://opensea.io/assets/matic/0x2953399124f0cbb46d2cbacd8a89cf0599974963/113334526430820078059276426234187275121195017550191774494822150233745037148736

... to another address, just as an experiment.

However, I'm getting this error that I can't seem to figure out:

```
/MaticTransfer/node_modules/web3-eth-abi/lib/index.js:297
        throw new Error('Returned values aren\'t valid, did it run Out of Gas? ' +
              ^

Error: Returned values aren't valid, did it run Out of Gas? You might also see this error if you are not using the correct ABI for the contract you are retrieving data from, requesting data from a block number that does not exist, or querying a node which is not fully synced.
    at ABICoder.decodeParametersWith (/MaticTransfer/node_modules/web3-eth-abi/lib/index.js:297:
```
Any ideas? Thanks in advance :grinning:

**Comment by imkane** [comment]
Is the issue that this token I'm attempting to transfer is ERC1155 instead of ERC721? :thinking: 

If so, that would make me quite silly lol.

If so, does anyone know a way to transfer ERC1155 tokens?

**Reply by ujjwalguptaofficial** [reply]
transfer erc1155 is not added in matic.js, just added a feature request - https://github.com/maticnetwork/matic.js/issues/273

**Comment by imkane** [comment, marked as answer]
@ujjwalguptaofficial thank you for implementing ERC1155 transfer (https://maticnetwork.github.io/matic.js/docs/pos/ERC1155/transferERC1155/), however I realized my `Returned values aren't valid, did it run Out of Gas?` error is coming from `getMaticPlasmaClient()` in https://github.com/maticnetwork/matic.js/blob/master/examples/utils.js.

Any ideas why that's happening and what it means?

This Matic stuff is incredibly confusing to me ... I'm not sure what to use for:

- network
- version
- parentProvider
- maticProvider

I simply want to be able to transfer ERC1155 tokens that are on OpenSea to another address :thinking:

**Reply by savannahar68** [reply]
@imkane did you figure out the issue? What was the fix?

**Reply by imkane** [reply]
@savannahar68 sadly I did not :cry: If you figure it out, please let me know!

**Reply by ujjwalgupta94** [reply]
try chaning the RPC, it's mostly due to RPC.

**Reply by savannahar68** [reply]
@ujjwalgupta94 You are right. The rpc was of testnet and network and version was of mainnet. Using proper config solved the issue thanks.

**Reply by imkane** [reply]
@savannahar68 can you please post your config? I'm not sure what to update lol.

**Reply by savannahar68** [reply]
@imkane go to `alchemy.com` and create MATIC RPC for mumbai region (If you want to try on testnet). Similarly from `infura.io` create `goerli` ROOT RPC. Use network as `testnet` and version as `mumbai`. Make sure the wallet you are using is on testnet.

This should be enough for setting up PlasmaClient using this library.
